src/types/subscriptionsTypes.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+export enum ChainSubscriptions {
+  SOURCE = 'sourceSubscriptions',
+  TARGET = 'targetSubscriptions'
+}
```

Comment:
I'm not a fan of the enum-object-choose-enum approach (it's next time we do it).
The way I see this long term is that we simply have:
```
const chainRelatedObjects = [a, b, c];
const chainIdx = 0;

chainRelatedObjects[chainIdx];
```

So in that case `SubscriptionsContextType` (and `SourceTargetState`) instead of being an object should be an array, and there should be one function that just gives you `chainIdx` based on the chain name and it's simply used to retreive the correct subscription context or api instance, etc.

That's something that avoid code duplication long term.

Reply:
hmmmm , interesting .. but how you would identify which connection plays the source and target role?  where would you put that logic?

Reply:
Well the selected account is associated with some chain name or chain index. And:
```js
const { selectedAccount } = useSelectedAccount();
const connections = useConnections();
const sourceChain = connections[selectedAccount.chainIndex];
// or even
/// const sourceChain = selectedAccount.chain
```

Reply:
i like the approach , i will take it into account.